### ISSUE TITLE
react/self-closing-comp rule is applied to HTML tags whose contents are markdown

### ISSUE THREAD

**Issue by Phebonacci** [mdx-js/eslint-mdx#547]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

3.1.5

### Link to runnable example

https://codesandbox.io/p/devbox/eslint-mdx-bug-rc3dvc?file=%2Fpackage.json%3A17%2C26

### Steps to reproduce

1. On the terminal, run eslint for the test file:
    ```
    yarn eslint app/test.mdx
    ```

### Expected behavior

The react/self-closing-comp rule should not trigger for the list tags whose contents are markdown.

### Actual behavior

The `react/self-closing-comp` rule is triggered for the list tags even though they have contents.
![image](https://github.com/user-attachments/assets/a596789b-2dd8-48c9-878e-e7980c78049a)

### Runtime

Node v20

### Package manager

yarn

### Build and bundle tools

Next.js

### Relevant issues

Something similar has been [reported before](https://github.com/mdx-js/eslint-mdx/issues/488) and it has been fixed. The difference with this new issue is the content is a markdown:
![image](https://github.com/user-attachments/assets/bfed6b5c-1299-4893-990d-f2122a87d918)
